### ISSUE TITLE
Linux tray icon is (still) not showing up: Cinnamon/Mint 18.2

### ISSUE THREAD

**Issue by nikhilw** [hluk/CopyQ#811]
I am facing an issue where the tray icon for Linux Mint 18.2 with Cinnamon is not showing up.

Here is how it looks like in tray:
![image](https://user-images.githubusercontent.com/1387025/31178398-8df6993c-a936-11e7-9ad3-faebfd0dd953.png)

Copyq version:
`
Me@Machine:~$ copyq --version
CopyQ Clipboard Manager v3.1.1 (hluk at email.cz)
Built with: Qt 5.5.1, LibQxt 0.7.0

`

**Comment by hluk** [comment]
Works fine for me on Ubuntu 16.04 with Cinnamon 2.8.6. Though my tray looks a bit different. Do you have any extensions for panel installed?

**Comment by nikhilw** [comment]
Good point, I am using the "Collapsible Tray" applet. I tried reverting to the default system tray and the icon did show up.
What is odd though is I used an older version till now, 2.8.3 I believe, but never had this issue (have always been using collapsible systray applet). It popped up after I switched to 3.1.1.

**Comment by hluk** [comment]
I just tested [Collapsible Systray](https://cinnamon-spices.linuxmint.com/applets/view/265) but I cannot add it to panel - it ends up with some error.

**Comment by nikhilw** [comment]
ohh.. It seems the cinnamon version on mint 18.2 is "3.4.6+sonya". There have been incompatible changes from 2.8, 3.2 etc for sure, I remember applets breaking. May be it was updated to support newer versions. Is there anything I can do to get you the logs or something?

**Comment by hluk** [comment]
Cannot reproduce it on Arch Linux, Cinnamon 3.4.6 -- tray works fine (with and without the extension).

Can you find logs for the extension?

**Comment by nikhilw** [comment]
When session starts the icon shows up, some time later it disappears. Similarly, if I killed and restarted copyq, the icon shows up for a few minutes, is visible on collapse-expand cycles, but some time later it disappears again. There does not seem anything in the logs; here are the logs: (<NOTE> are my comments in logs)

```
info t=2017-10-10T10:29:30Z [collapsible-systray@feuerfuchs.eu] Insert menu item for slack in active applications
<NOTE: the icon had disappeared>
info t=2017-10-10T10:29:42Z [collapsible-systray@feuerfuchs.eu] Unregister instance of copyq
<NOTE: copyq stopped and restarted>
info t=2017-10-10T10:29:42Z [collapsible-systray@feuerfuchs.eu] No more instances left
info t=2017-10-10T10:29:42Z [collapsible-systray@feuerfuchs.eu] Insert menu item for copyq in inactive applications
info t=2017-10-10T10:32:30Z Adding systray: copyq (24x24px)
info t=2017-10-10T10:32:31Z [collapsible-systray@feuerfuchs.eu] Event: _insertStatusItem - copyq
info t=2017-10-10T10:32:31Z [collapsible-systray@feuerfuchs.eu] Register instance of copyq
info t=2017-10-10T10:32:31Z [collapsible-systray@feuerfuchs.eu] Insert menu item for copyq in active applications
<NOTE: the icon shows up, visible on multiple collapse-expand cycles>
<NOTE: ~15 mins or so>
<NOTE: the icon has disappeared, no update in logs>
```

**Comment by nikhilw** [comment]
To test, reverted to 2.8.3, been using it for hours, the icon issue does not seem to occur with this version.

**Comment by nikhilw** [comment]
I agree. I went up trying each version from there. It worked fine till 3.0.3, issue starts with version: 3.1.0; Logs same as above, basically nothing after 'insert menu item for copyq' and a few minutes later the icon is gone again.

```
info t=2017-10-10T23:06:46Z [collapsible-systray@feuerfuchs.eu] Unregister instance of copyq
info t=2017-10-10T23:06:46Z [collapsible-systray@feuerfuchs.eu] No more instances left
info t=2017-10-10T23:06:46Z [collapsible-systray@feuerfuchs.eu] Insert menu item for copyq in inactive applications
info t=2017-10-10T23:07:14Z Adding systray: copyq (24x24px)
info t=2017-10-10T23:07:15Z [collapsible-systray@feuerfuchs.eu] Event: _insertStatusItem - copyq
info t=2017-10-10T23:07:15Z [collapsible-systray@feuerfuchs.eu] Register instance of copyq
info t=2017-10-10T23:07:15Z [collapsible-systray@feuerfuchs.eu] Insert menu item for copyq in active applications
```

**Comment by nikhilw** [comment]
Just writing to confirm that the issue seems resolved with version v3.1.2, the icon stays visible.
Thank you @hluk for your time and effort into this.
